chapter13/dtj/ex8

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+print "Give me a directory\n";
+chomp (my $directory = <STDIN>);
+
+if ($directory =~ /\S/ ) {
+  chdir $directory or die "$! error when trying to open $directory";
+} else {
+  print "Only whitespace supplied, attempting home directory\n";
+  chdir or die "$! error when trying to open home directory";
+}
+
+foreach my $file (sort <.* *>) {
+  my $dest = readlink $file;
+  print "$file -> $dest\n" if (-l $file);
```

Comment:
$dest should be undef from readlink if its is not a link, might be a more efficient test here than -l

Then again, might not.